File: typescript/src/models/cita.ts
import { Doctor } from './doctor';
import { ModalidadCita } from '../DS/modalidadCita';
import { StatusConsulta } from '../DS/statusConsulta';
import { Paciente } from './paciente';

export class Cita {
    private fechaHora?: Date;
    private modalidad: ModalidadCita;
    private status: StatusConsulta;
    private doctor: Doctor;
    private paciente: Paciente;

    constructor(modalidad: ModalidadCita, status: StatusConsulta, doctor: Doctor, paciete: Paciente, fechaHora?: Date) {
        this.fechaHora = fechaHora;
        this.modalidad = modalidad;
        this.status = status;
        this.paciente = paciete;
        this.doctor = doctor;
    }

    public getFechaHora(): Date | undefined {
        return this.fechaHora;
    }

    private setFecha(fechaHora: Date) {
        this.fechaHora = fechaHora;
    }

    public getModalidad(): ModalidadCita {
        return this.modalidad;
    }

    private setModalidad(modalidad: ModalidadCita) {
        this.modalidad = modalidad;
    }

    public getStatus(): StatusConsulta {
        return this.status;
    }

    private setStatus(status: StatusConsulta) {
        this.status = status;
    }

    public getDoctor(): Doctor {
        return this.doctor;
    }

    private setDoctor(doctor: Doctor) {
        this.doctor = doctor;
    }

    public getPaciente(): Paciente {
        return this.paciente;
    }

    private setPaciente(paciente: Paciente) {
        this.paciente = paciente;
    }
}

File: dart/lib/src/models/cita.dart
import 'doctor.dart';
import 'paciente.dart';
import 'fecha.dart';
import '/src/enum/cita.dart';

class Cita {
  Fecha fecha;
  Hora hora;
  Modalidad modalidad;
  StatusConsulta status;
  Doctor doctor;
  Paciente paciente;
  Cita(this.fecha, this.hora, this.modalidad, this.status, this.doctor,
      this.paciente);

  Fecha get getFecha {
    return this.fecha;
  }

  void set setFecha(Fecha fec) {
    this.fecha = fec;
  }

  Hora get getHora {
    return this.hora;
  }

  void set setHora(Hora hor) {
    this.hora = hor;
  }

  Doctor get getDoctor {
    return this.doctor;
  }

  void set setDoctor(Doctor doc) {
    this.doctor = doc;
  }

  Modalidad get getModalidad {
    return this.modalidad;
  }

  void set setModalidad(Modalidad mod) {
    this.modalidad = mod;
  }

  StatusConsulta get getStatus {
    return this.status;
  }

  void set setStatus(StatusConsulta sta) {
    this.status = sta;
  }

  Paciente get getPaciente {
    return this.paciente;
  }
}
